### DISCUSSION TITLE
Problem with texture in gltf models

### DISCUSSION THREAD

**Discussion by tomasbym** [pyvista/pyvista#1643] [Q&A]
Hi,
I am trying to use pyVista for visualizing gltf models in Holoviz Panel framework. My problem might not be directly related to pyVista but somebody could hopefully point me to the right direction. My problem is that when I load a model with texture using pyVista, it shows correctly in the build in viewer but when I try to visualize the same model in Panel view using same renderer the texture is missing.
Here is a simple code to test.
Thanks, T.

```
import pyvista
import panel as pn
from pyvista import examples

helmet_file = examples.gltf.download_damaged_helmet()
pl = pyvista.Plotter()
pl.import_gltf(helmet_file)
#using pyvista viewer
pl.show()
#using Panel view - comment the line above
vtkpan = pn.panel(pl.ren_win)
pn.serve(vtkpan, threaded=True)
```

**Comment by akaszynski** [comment]
Looks like the texture isn't being transferred properly into the panel viewer.  This is something we're going to have to look into, but I don't think it will be an easy fix (and might be easier with pythreejs with #1557).

**Reply by tomasbym** [reply]
Thanks for your answer. 
I have been trying to use pythreejs embedded in bokeh IPyWidget but without any luck. Could you please tell why this example is not working?
T.

```
import pyvista as pv
import panel as pn
from pyvista import examples
from ipywidgets_bokeh import IPyWidget
from pyvista.jupyter.notebook import handle_plotter

helmet_file = examples.gltf.download_damaged_helmet()
pl = pv.Plotter()#off_screen=True)

scene = handle_plotter(pl, backend='ipygany', return_viewer=True)

widget = IPyWidget(widget=scene)
vtkpan = pn.panel(widget)
pn.serve(vtkpan, threaded=True)
```

**Reply by akaszynski** [reply]
Looks like the backend is still ipygany in this example.  You'll need to either wait for #1557 to be merged and released in 0.32.0, or check the `fix/ipygany_structured_mesh` branch.

I can say that it's still going to be broken since I've not mapped over the physically based rendering textures over to three.js as of yet.  It's non-trivial and I doubt will be patched in panel (which uses vtk.js).

**Reply by tomasbym** [reply]
Is the "mapping of the physically based rendering over to three.js" something that you plan to implement? If so, could you please estimate  if that would be in "near" or "far" future. 
Thanks for your contributions to this great opensource project.

**Reply by akaszynski** [reply]
Yes, just wasn't able to get it in 0.32.0.  Textures are hard...
